libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs

```diff
@@ -28,14 +28,14 @@ public class TestFlow
     {
         private readonly TestAdapter _adapter;
         private readonly Task _testTask;
-        private Func<ITurnContext, Task> _callback;
+        private BotCallbackHandler _callback;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="TestFlow"/> class.
         /// </summary>
         /// <param name="adapter">The test adapter to use.</param>
         /// <param name="callback">The bot turn processing logic to test.</param>
-        public TestFlow(TestAdapter adapter, Func<ITurnContext, Task> callback = null)
+        public TestFlow(TestAdapter adapter, BotCallbackHandler callback = null)
```

Comment:
I know you've done all the work already so this might be moot, but considering most tests will not do anything with the `CancellationToken`, you could create a constructor overload that looks like this to keep test callbacks from having to deal with the token:

```CSharp
public TestFlow(TestAdapter adapter, Func<ITurnContext, Task> callback) 
    : this(adapter, callback == null ? null : (tc, ct) => callback(tc))
{
}
```

It just makes it easier/cleaner to write test callbacks. If you choose to do this you can revert all the updates you had to make in this PR to deal with the change as well.